File: 2020_2021/L3_S5_2020-2021/INFO0503_modélisation web serv & web avancé/INFO503/TP1-2/src/Usines/Place.java
package Usines;

import Voiture.Voiture;

public class Place {
	private char rangee;
	private int num;
	private boolean occupation;
	private Voiture voiture;
	
	
	public Place(char rangee, int num) {
		this.rangee = rangee;
		this.num = num;
		this.occupation = false;
		this.voiture = null;
	}

	public Place(char rangee, int num, Voiture voiture) {
		this.rangee = rangee;
		this.num = num;
		this.voiture = voiture;
		this.occupation = true;
	}
	
	public Place(Place pl) {
		this.rangee = pl.rangee;
		this.num = pl.num;
		this.occupation = pl.occupation;
		this.voiture = pl.voiture;
	}
	
	public char getRangee() {
		return rangee;
	}
	public int getNum() {
		return num;
	}
	public boolean isOccupation() {
		return occupation;
	}
	public Voiture getVoiture() {
		return voiture;
	}
	
	public void setVoiture(Voiture voiture) {
		this.voiture = voiture;
		this.occupation = true;
	}
	
	/*
	public void setOccupation(boolean occupation) {
		this.occupation = occupation;
	}
	*/
	
	public Voiture enleverVoiture() {
		if(!this.occupation) return null;
		Voiture voiture = this.getVoiture();
		this.voiture = null;
		this.occupation = false;
		return voiture;
	}
	
	@Override
	public String toString() {
		return "Place [rangee=" + rangee + ", num=" + num + ", occupation=" + (this.occupation?(this.occupation+ " ; Voiture : " + this.voiture.toString() + "]"):(this.occupation + "]"));
	}
	
}

File: 2020_2021/L3_S5_2020-2021/INFO0503_modélisation web serv & web avancé/INFO503/TP2/Usine/Place.php
<?php
class Place{
    protected $rangee;
    protected $numPlace;
    protected $occupee;
    protected $voiture;

    public function __construct($rangee = 'A', $numPlace = 0, $voiture = null){
        $this->rangee = $rangee;
        $this->numPlace = $numPlace;
        if($voiture != null){
            $this->occupee = true;
            $this->voiture = $voiture; 
        }
        else{
            $this->occupee = false;
            $this->voiture = null;
        }
    }

    public function getRangee() : String {return $this->rangee;}
    public function getNumPlace() : int {return $this->numPlace;}
    public function getOccupee() : bool {return $this->occupee;}
    public function getVoiture() {return $this->voiture;}

    public function setVoiture(Voiture $voiture = null){
        if($voiture != null){
            $this->occupee = true;
            $this->voiture = $voiture; 
        }
        else{
            $this->occupee = false;
            $this->voiture = null;
        }
    }
    public function setOccupee(bool $bool = false){
        if($this->getVoiture() != null ){echo("la place que vous avez sélectionnée est occupée.");}
        else{$this->occupee = $bool; echo("Voiture mise en place.");}
    }

//Méthodes
    public function __toString() : String{
        $occupation = ($this->getOccupee()) ? ($this->afficherBoolean($this->getOccupee()) . " Voiture : " . $this->getVoiture()->__toString()) : ($this->afficherBoolean($this->getOccupee()));
        return "rangee : " . $this->getRangee() . " num : " . $this->getNumPlace() . " occupee : " . $occupation;
    }

    public static function enleverVoiture($place){        
        $voiture = $place->getVoiture();
        if($place->getOccupee()){
            $place->voiture = null;
            $place->occupee = false;
        }
        return $voiture;
    }

    private function afficherBoolean(bool $boolean) : String{
        $affichage = ($boolean) ? (1) : (0);
        return $affichage;
    }
}